### PR TITLE
refactor: share worker bundling between both `publish` and `dev` commands

### LINKED ISSUE #401: Add deprecation warning when trying to run commands without explicit entrypoint
In wrangler1, you could run `wrangler dev`, `wrangler publish`, etc. without explicitly specifying the entrypoint in the command. Many users will attempt to run these old commands and get confused why they now see the help page. Instead, there should be a deprecation warning that tells them what to do.

### PR DESCRIPTION
**Pull request by petebacondarwin**
This changes moves the code that does the esbuild bundling into a shared file
and updates the `publish` and `dev` to use it, rather than duplicating the
behaviour.

See #396
Resolves #401

### CODE REVIEW

File: packages/wrangler/src/dev.tsx
Comment (JacobMGEvans) [inline packages/wrangler/src/dev.tsx:563]: saw further down, used this style. Change not necessary. 
```suggestion
        modules,
```

**Comment by petebacondarwin** [comment]
I have manually tested the `dev` and `publish` for the following scenarios:

- simple worker dev remote
- simple worker dev local
- simple worker publish
- sites worker dev remote
- sites worker dev local
- sites worker publish
- experimental-public worker dev remote
- experimental-public worker dev local
- experimental-public worker publish

Sooooo.... I believe this PR will not break anything. (Famous last words).

**Comment by JacobMGEvans** [comment]
I am happy to approve once the conflicts are resolved.

**Comment by petebacondarwin** [comment]
I have rebased on the change that added the `format` parameter to the build step.
I'll wait for a 👍  from @threepointone before merging.

**Review by threepointone — APPROVED** [review]
lfggggg

doing the refactor robot dance

**Pull request merged**
